### ISSUE TITLE
Catboost predict out of range value

### ISSUE THREAD

**Issue by tavoletta** [catboost/catboost#1239]
Problem: Hi, I am using a catboost model to predict a target that is a ratio (0-1 values). Sometimes, I get negative values. Is there a way to force catboost to predict inside a given range? Thanks
catboost version:
Operating System:
CPU:
# GPU:

**Comment by mik3githubber** [comment]
If this is a regression problem it's not going to work out of the box.  In the past, for this time of problem I would either:

- manually clip values less than 0 or greater than 1
- Run it as a classfication problem with softmax output (your classes could be percentages say 0%, 1% through to 100%).  You could take a weighted average of the softmax output.  This is computationally quite expensive given the number of classes
-Run it as a binary classification problem with your Y value ratios being the target (as opposed to just 0 and 1). Your values will be rangebound here.

**Comment by annaveronika** [comment]
It is not possible to restrict values to a specified range when using RMSE regression.
As @mik3githubber suggested, you can clip values less than 0 or greater than 1.
Other option: if your values are probabilities, you can use classification in with loss_function='CrossEntropy', this will give you probabilities.

Hope it helps, closing the issue.

**Comment by razamit** [comment]
Hi

We are facing a similar issue.
Our regression labels are -10 to 10 with gaussian distribution. The problem is that our model predicts values in a much smaller range for example 0 to 0.4. Its not even close to the range of the labels.

what can cause this? 

Thanks
Amit
